### PR TITLE
Fixes the behavior of TC tree size counting so as to avoid triggering…

### PR DESCRIPTION
**Pull request by barath**
… a race.

### CODE REVIEW

**Comment by melvinw** [comment]
Looks good. Just one question. Why store sizes in `TrafficClassBuilder::tc_count` instead of as a member of `TrafficClass`?

File: core/traffic_class.h [hunk 1]
```diff
@@ -124,13 +124,9 @@ class TrafficClass {
...
 
-  size_t Size() const {
-    size_t sz = 0;
-    Traverse([](const TrafficClass *,
-                void *arg) { *reinterpret_cast<size_t *>(arg) += 1; },
-             static_cast<void *>(&sz));
-    return sz;
-  }
+  // Returns the number of TCs in the TC subtree rooted at this, including this
+  // TC.  Returns 0 upon error.
```
Comment (sangjinhan) [inline core/traffic_class.h]: Is it ever possible (in normal situations) to return 0? If not, shouldn't it be CHECK() or DCHECK()?

**Comment by barath** [comment]
I didn't store it in TrafficClass to minimize unnecessary things being held
in TrafficClass objects which are on the fast path.

File: core/traffic_class.h (hunk 1 above)
Comment (barath) [inline core/traffic_class.h]: Agreed.  Would be happy to change it to CHECK.

**Comment by melvinw** [comment]
Gotcha. Fine by me.

File: core/traffic_class.h (hunk 1 above)
Comment (sangjinhan) [inline core/traffic_class.h]: Yes please 👍

**Comment by melvinw** [comment]
Thanks! 
5ih45x7

**Pull request merged**